scanpipe/pipes/d2d.py

```diff
@@ -1343,3 +1343,31 @@ def flag_deployed_from_resources_with_missing_license(project, doc_extensions=No
 
     no_license_files.update(status=flag.NO_LICENSES)
     unknown_license_files.update(status=flag.UNKNOWN_LICENSE)
+
+
+def handle_dangling_deployed_legal_files(project, logger):
+    to_resources = project.codebaseresources.files().to_codebase().no_status()
+    legal_files = [resource for resource in to_resources if resource.is_legal]
```

Comment:
This seems highly inefficient, you should use a QuerySet.

Reply:
That's the reason it's a workaround, as of now we don't have `is_legal` as a field in our model and I don't see how we can use model property in QuerySet.

Reply:
That's the point, do not use a property, create a QuerySet method instead.

Reply:
Thanks @tdruez 
We're now using QuerySet insted property :)